### PR TITLE
Bug fixes, cleanup and Node/NPM package upgrades to current dependency versions.

### PR DESCRIPTION
**Pull request by grempe**

### CODE REVIEW

**Comment by grempe** [comment]
I think this is failing just due to node version check in check_dependencies.sh

I'll add a commit to test for either version.

**Comment by daviddahl** [comment]
oh wow!

**Comment by devgeeks** [comment]
:100: :tada: :joy_cat:

**Comment by grempe** [comment]
OK, anyone know how you run these integration tests in the PhantomJS context now? The coffee script file referenced in the Makefile no longer exists.

````
test-integration: node_modules kill-server reset build-client run-server
	@echo "Running integration tests..."
	#@./node_modules/phantomjs/bin/phantomjs --ignore-ssl-errors=true ./node_modules/mocha-phantomjs/lib/mocha-phantomjs.coffee test/index.html; SUCCESS=$$?; \
	$(MAKE) kill-server; \
	exit $$SUCCESS
````

**Comment by grempe** [comment]
If I had my way, this would get merged as is even though I had to temporarily disable the top level integration tests.  Would like to avoid a later rebase since there are a bunch of commits in here.  Lets figure out how to get the integration tests working (in browser and in phantomJS) right after. Work for you?

**Comment by grempe** [comment]
Making very good progress getting the integration tests to run in a browser.  Less progress so far getting them to run in phantoms headless.  Working on it.

File: server/app.js [hunk 1]
```diff
@@ -112,29 +118,39 @@ app.use(expressWinston.logger({
...
 }));
 
 app.config = require('./lib/config');
-logger.info("app.config", app.config);
+
+// logger.info('app.config', app.config);
 
 app.datastore = require('./lib/storage');
 
+var redisSession = require('./redis-session')({
```
Comment (daviddahl) [inline server/app.js]: Hotness

File: server/routes/account.js [hunk 2]
```diff
@@ -225,33 +241,3 @@ app.post('/account/:username/keyring',
     });
   }
 );
-
-/**!
- * ### GET /accountBuffer
- * If there is a maximumUsers configuration variable set,
- * return the amount of free accounts - otherwise, return null.
-*/
-app.get('/accountBuffer', function (req, res) {
```
Comment (daviddahl) [inline server/routes/account.js]: We don't currently use this but are you removing it for a reason? (maybe it re-surfaces later in this PR?)

**Comment by daviddahl** [comment]
Never mind about the user limit comment. Looks great! Works well. Thank you!

**Pull request merged**